csFastFloat/Utils/Utils.cs

```diff
@@ -213,5 +217,64 @@ internal unsafe static bool strncasecmp(byte* input1, ReadOnlySpan<byte> input2,
 
     
 
+    [MethodImpl(MethodImplOptions.AggressiveInlining)]
+    public static int LeadingZeroCount(ulong value)
+    {
+#if HAS_BITOPERATIONS
+      return System.Numerics.BitOperations.LeadingZeroCount(value);
+#else
+      uint hi = (uint)(value >> 32);
+ 
+      if (hi == 0)
+      {
+        return 32 + Log2SoftwareFallback((uint)value);
+      }
+ 
+      return Log2SoftwareFallback(hi);
+      
+      [MethodImpl(MethodImplOptions.AggressiveInlining)]
+      static int Log2SoftwareFallback(uint value)
+      {
+       if (value == 0)         return 32;
+      int n = 1;
+      if (value >> 16 == 0) { n += 16; value <<= 16; }
+      if (value >> 24 == 0) { n +=  8; value <<=  8; }
+      if (value >> 28 == 0) { n +=  4; value <<=  4; }
+      if (value >> 30 == 0) { n +=  2; value <<=  2; }
+      n -= (int) (value >> 31);
+      return n;
```

Comment:
I did not validate this algorithm (I am going to trust you). It seems reasonable. It is not going to be fast, but that's fine.

Reply:
I would prefer to understand why the previous version wasn't working as it should...  at least we've got a workaround for now.

Reply:
I would not waste a lot of time on it. If you are convinced that it is something important, just go with what works, and create an issue explaining your concerns. Someone might pick it up. If not, your solution is fine. I would absolutely release 2.0 with this new code of yours.